Source/WTF/wtf/text/cf/TextBreakIteratorCFCharacterCluster.h

```diff
@@ -66,30 +68,31 @@ class TextBreakIteratorCFCharacterCluster {
             return { };
         if (location > m_stringLength)
             return m_stringLength;
-        auto range = CFStringGetRangeOfCharacterClusterAtIndex(m_string.get(), location - 1, m_type);
-        return range.location;
+        auto range = CFStringGetRangeOfCharacterClusterAtIndex(m_string.get(), location - 1 + m_priorContextLength, m_type);
+        return range.location - m_priorContextLength;
```

Comment:
This needs to be `std::max(static_cast<unsigned long>(range.location), m_priorContextLength) - m_priorContextLength;`